docs/datastorage/capabilities.md

```diff
@@ -30,27 +30,27 @@ Here is an example of discouraged capability usage:
 
 ## NeoForge-provided capabilities
 
-NeoForge provides capabilities for the following three interfaces: `IItemHandler`, `IFluidHandler` and `IEnergyStorage`.
+NeoForge provides capabilities for the following three interfaces: `ResourceHandler<ItemResource>`, `ResourceHandler<FluidResource>` and `EnergyHandler`.
 
-`IItemHandler` exposes an interface for handling inventory slots. The capabilities of type `IItemHandler` are:
+`ResourceHandler<ItemResource>` exposes an interface for handling inventory slots. The capabilities of type `ResourceHandler<ItemResource>` are:
 
-- `Capabilities.ItemHandler.BLOCK`: automation-accessible inventory of a block (for chests, machines, etc).
-- `Capabilities.ItemHandler.ENTITY`: inventory contents of an entity (extra player slots, mob/creature inventories/bags).
-- `Capabilities.ItemHandler.ENTITY_AUTOMATION`: automation-accessible inventory of an entity (boats, minecarts, etc).
-- `Capabilities.ItemHandler.ITEM`: contents of an item stack (portable backpacks and such).
+- `Capabilities.Item.BLOCK`: automation-accessible inventory of a block (for chests, machines, etc).
+- `Capabilities.Item.ENTITY`: inventory contents of an entity (extra player slots, mob/creature inventories/bags).
+- `Capabilities.Item.ENTITY_AUTOMATION`: automation-accessible inventory of an entity (boats, minecarts, etc).
+- `Capabilities.Item.ITEM`: contents of an item stack (portable backpacks and such).
 
-`IFluidHandler` exposes an interface for handling fluid inventories. The capabilities of type `IFluidHandler` are:
+`ResourceHandler<FluidResource>` exposes an interface for handling fluid inventories. The capabilities of type `ResourceHandler<FluidResource>` are:
 
-- `Capabilities.FluidHandler.BLOCK`: automation-accessible fluid inventory of a block.
-- `Capabilities.FluidHandler.ENTITY`: fluid inventory of an entity.
-- `Capabilities.FluidHandler.ITEM`: fluid inventory of an item stack.
-This capability is of the special `IFluidHandlerItem` type due to the way buckets hold fluids.
+- `Capabilities.Fluid.BLOCK`: automation-accessible fluid inventory of a block.
+- `Capabilities.Fluid.ENTITY`: fluid inventory of an entity.
+- `Capabilities.Fluid.ITEM`: fluid inventory of an item stack.
+    - This capability specifies an `ItemAccess` context due to the way buckets hold fluids.
```

Comment:
This is not exclusive to fluids. `Capabilities.Item.ITEM` and `Capabilities.Energy.ITEM` also specify an `ItemAccess` context.